### ISSUE TITLE
Cell bodies with "battery body alignment"

### ISSUE THREAD

**Issue by jeffreykegler** [jeffreykegler/yahc#23]
Below is an extract of `hoon.hoon`, which shows what I am calling "battery alignment" of cell bodies.  We allow something similar for jogs -- instead of the tail following the head by one stop, a jogging can align its tails with each other.

But in battery cells this style is restricted to a very few places.  `hoon.hoon` has these few, 87 occur in `hood.hoon`, and 278 in `old-zuse.hoon`.

The question: Allow battery alignment of cell bodies, as exemplified below?  Or add these cases to `anomaly.suppressions`?

```
::::  4b: text processing
  ::
++  at                                                  ::  basic printing
  |_  a/@
  ++  r
    ?:  ?&  (gte (met 3 a) 2)
            |-
            ?:  =(0 a)
              &
            =+  vis=(end 3 1 a)
            ?&  ?|(=('-' vis) ?&((gte vis 'a') (lte vis 'z')))
                $(a (rsh 3 1 a))
            ==
        ==
      rtam
    ?:  (lte (met 3 a) 2)
      rud
    rux
  ::
  ++  rf    `tape`[?-(a $& '&', $| '|', * !!) ~]
  ++  rn    `tape`[?>(=(0 a) '~') ~]
  ++  rt    `tape`['\'' (weld (mesc (trip a)) `tape`['\'' ~])]
  ++  rta   rt
  ++  rtam  `tape`['%' (trip a)]
  ++  rub   `tape`['0' 'b' (rum 2 ~ |=(b/@ (add '0' b)))]
  ++  rud   (rum 10 ~ |=(b/@ (add '0' b)))

```

**Comment by ohAitch** [comment]
Hm, how many unaligned same-line arm bodies are there? I'm guessing a lot of them are _naturally_ aligned by every single arm in the core being four(or three) letters long, though that will be less the case in newer code.

old-zuse.hoon in particular is a shim for getting old code to compile after a refactoring / guide for doing the translation, and is iirc gone in 2019 master, so if it's the main user(and there's a lot of unaligned code) probably just add these to anomalies yeah

**Comment by jeffreykegler** [comment]
There are 670 batteries in the `arvo/` corpus, of which 302 have all their joined cells aligned at the same column.  280 do not have a defined alignment status, because all of their cells are split.  And 88 are unaligned.  The 88 would include batteries intended to be aligned, but with errors, which would mean an unalignment rate of 10-20%.

It might make sense to make batteries orthogonal with our treatment of joggings.  There we found a battery-wide body alignment wherever possible, and treated a cell body as properly aligned if it either matched the battery-wide alignment, or was exactly one stop after the head.  It's a standard that gives the writer latitude to do sensible things, but still stays fairly strict.

@ohAitch -- let me know what you think.

**Comment by ohAitch** [comment]
Yup that makes sense. How many of the unaligned arms are "truly ragged",
i.e. at least one of the body columns is different, vs "ragged aligned",
where all joined arm names are of the same length and thus the bodies are
/both/ exactly one stop past and on the same column?

**Comment by jeffreykegler** [comment]
As I counted it this time, all 88 of the "unaligned" were truly ragged.

**Comment by jeffreykegler** [comment]
Implemented battery body alignment as of commit b267979 .  Will mark this "ready to be closed"

**Comment by jeffreykegler** [comment]
Closed as per the above.
